exp/api/v1beta1/gcpmanagedmachinepool_types.go

```diff
@@ -54,6 +83,57 @@ type GCPManagedMachinePoolSpec struct {
 	ProviderIDList []string `json:"providerIDList,omitempty"`
 }
 
+// NodeNetworkConfig encapsulates node network configurations.
+type NodeNetworkConfig struct {
+	// Tags is list of instance tags applied to all nodes. Tags
+	// are used to identify valid sources or targets for network
+	// firewalls.
+	// +optional
+	Tags []string `json:"tags,omitempty"`
+	// CreatePodRange specifies whether to create a new range for
+	// pod IPs in this node pool.
+	// +optional
+	CreatePodRange *bool `json:"createPodRange,omitempty"`
+	// PodRangeName is ID of the secondary range for pod IPs.
+	// +optional
+	PodRangeName *string `json:"podRangeName,omitempty"`
+	// PodRangeCidrBlock is the IP address range for pod IPs in
+	// this node pool.
+	// +optional
+	PodRangeCidrBlock *string `json:"podRangeCidrBlock"`
+}
+
+// NodeSecurityConfig encapsulates node security configurations.
+type NodeSecurityConfig struct {
+	// ServiceAccount specifies the identity details for node
+	// pool.
+	// +optional
+	ServiceAccount ServiceAccountConfig `json:"serviceAccount,omitempty"`
+	// SandboxType is type of the sandbox to use for the node.
+	// +optional
+	SandboxType *string `json:"sandboxType,omitempty"`
+	// EnableSecureBoot defines whether the instance has Secure
+	// Boot enabled.
+	// +optional
+	EnableSecureBoot *bool `json:"enableSecureBoot,omitempty"`
+	// EnableIntegrityMonitoring defines whether the instance has
+	// integrity monitoring enabled.
+	// +optional
+	EnableIntegrityMonitoring *bool `json:"enableIntegrityMonitoring,omitempty"`
+}
+
+// ServiceAccountConfig encapsulates service account options.
+type ServiceAccountConfig struct {
```

Comment:
If you are specifying the service account config in `NodeSecurityConfig` then do you have to supply one of the fields in this struct?

Reply:
You can either specify `Email` which is Google Cloud Platform Service Account to be used, or specify `Scopes` for default service account (email of GKE default account or email field left blank).